packages/playground/src/components/form-inputs/ArrayInput.tsx

```diff
@@ -0,0 +1,95 @@
+import React, { useCallback } from 'react'
+import { Controller, useFieldArray } from '@harnessio/forms'
+import { Button, FormField } from '@harnessio/canary'
+import { Plus, Trash } from '@harnessio/icons-noir'
+import { InputType } from './types'
+import { InputComponent, InputProps } from '@harnessio/forms'
+import { AnyFormikValue, IInputDefinition } from '@harnessio/forms'
+import { RenderInputs } from '@harnessio/forms'
+import InputLabel from './common/InputLabel'
+import InputWrapper from './common/InputWrapper'
+import { InputError } from './common/InputError'
+
+export type UIInputWithConfigsForArray = Omit<IInputDefinition, 'path'>
+
+export interface ArrayInputConfig {
+  inputType: InputType.array
+  inputConfig: {
+    input: IInputDefinition
+  }
+}
+
+function ArrayInputInternal(props: InputProps<AnyFormikValue, ArrayInputConfig>): JSX.Element {
+  const { readonly, path, input, factory } = props
+  const { label, required, inputConfig, description } = input
+
+  const { fields, append, remove } = useFieldArray({
+    name: path
+  })
+
+  const getChildInputs = useCallback(
+    (rowInput: UIInputWithConfigsForArray, parentPath: string, idx: number): IInputDefinition[] => {
+      const retInput = {
+        ...rowInput,
+        // NOTE: create absolute path using parent path and index
+        path: `${parentPath}[${idx}]`
+      } as IInputDefinition
+
+      return [retInput]
+    },
+    []
+  )
+
+  return (
+    <InputWrapper>
+      <FormField
+        name={path}
+        render={() => (
+          <>
+            <InputLabel label={label} required={required} description={description} />
+            <Controller
+              name={path}
+              render={() => (
+                <div className="flex flex-col">
+                  <div>
+                    {fields.map((_item, idx) => (
+                      <div key={idx} className="flex items-end space-x-2">
+                        {inputConfig?.input && (
+                          <RenderInputs items={getChildInputs(inputConfig?.input, path, idx)} factory={factory} />
+                        )}
+                        <div>
```

Comment:
is this extra div needed?
ideally the trash button should appear inline with the input field, with some standard spacing between them.